scripts/auth_collections.sh

```diff
@@ -220,12 +224,16 @@ function create_resource {
       json=$(jo id="$id" name="$name" description="test team for $name")
       resource=$RESOURCE ;;
     team-members)
-      # just add a 100 members to each team
+      # just add 100 members to each team
       json=$(jq -n --argjson n 100 'reduce range(1; $n) as $i (.; .user_ids += ["test-user-\($i)"])')
       resource="teams/test-teams-$id_index/users:add" ;;
     policies)
       json=$(jo subjects="$(jo -a token:test-tokens-$id_index)" action=read resource="compliance:*")
       resource=$RESOURCE ;;
+    rules)
+      # note: requires adding "foo-project" first
+      json=$(jo id="$id" name="$name" type=NODE project_id=foo-project conditions='[{"operator":"MEMBER_OF","attribute":"CHEF_SERVERS","values":["prod","staging"]}]')
```

Comment:
This is fine, just pointing out that the conditions bit could have been `jo conditions=$(jo -a $(jo operator=MEMBER_OF ...))`, too, I think.

Reply:
Yes, but shellcheck demands each nested `jo` be quoted... that's why I opted for this. 🤷‍♂

Reply:
hmm maybe `jo conditions="$(jo -a "$(jo operator=MEMBER_OF ...)")"` then. 🤷‍♂

Reply:
Right but you still have to escape the quotes, I would think, no?

Reply:
Have you tried? I don't think so, the way this line is formed. In bash, `"$(...)"` is not just a string.

Reply:
https://explainshell.com/explain?cmd=json%3D%22%24%28jo+id%3D%22%24id%22+name%3D%22%24name%22+type%3DNODE+project_id%3Dfoo-project+conditions%3D%22%24%28jo+-a+%22%24%28jo+operator%3DMEMBER_OF+attribute%3DCHEF_SERVERS+values%3D%22%24%28jo+-a+prod+staging%29%22%29%22%29%22%29%22#

Doesn't explain as much as I'd have hoped it would explain, but at least shows how those nested `"$(...)"` are valid. Tossing this at shellcheck.net shows no relevant errors. 😉

Reply:
Ah, cool. And thank you for the neat explainshell.com link!